hack/local-development/local-garden/start.sh

```diff
@@ -42,6 +42,9 @@ for i in 1..10; do
   echo "# Waiting until Kube-Apiserver is available"
 done
 
+echo "# Configuring RBAC resources for Gardener components"
+$(dirname $0)/configure-rbac
+
```

Comment:
I don't know how to achieve this for the `remote-garden`, unfortunately. In this case, we don't control the Kubernetes cluster and do not have access to the CA certificate in order to issue new client certs for the Gardener components. While we could use e.g. `ServiceAccount`s for GAPI,GCM,GSCHED,GAC, we have to use a client cert for gardenlet because of #1723 and how the seed identification works (not achievable with `ServiceAccount`s).

Reply:
A `remote-garden` is usually just a shoot in our `dev` landscape, or in the general case a cluster that the developer owns. In this case, the developer also can get hold of the CA certificate and e.g. pass it as a parameter to a script, or am I missing something?

Reply:
We discussed this and it became clear that even though it would be possible, it won't be used much and it would be inconsistent for `gardenlet` due to the requirement to have the `Webhook` authorization flag. So let's not do it here, it could be added later if there is a strong need.

Reply:
As discussed via phone, let's not invest into the remote garden setup for now because the way forward is not exactly clear, especially whether the effort we'd put is worth the benefits (this is also with the assumption that the local-garden is the default/most common development environment).
I added a disclaimer for the remote garden with 81041e3.